### PR TITLE
Track page views and searches in DSpace with a custom Angulartics2 provider

### PR DESCRIPTION
**Pull request by antoine-atmire**
In https://github.com/DSpace/dspace-angular/pull/241 support has been added for Google Analytics through the means of the Angulartics2 provider.
This PR adds another Angulartics2 provider to track page views and search request within DSpace itself.

What's included?
- A service that makes use of the [the endpoints](https://github.com/DSpace/Rest7Contract/blob/master/statistics.md) for tracking statistics within DSpace. 
- An Angulartics2 provider which makes use of this service.
- A couple components that are included on the search page, the item pages, collection pages, community pages and on the home page. These components will fire Angulartics2 events when rendered and those events are then picked up by the plugins. The components themselves are invisible.
- A SiteDataService and the related model classes. This is necessary to track a view request on the home page.

As with the Google Analytics plugin, the Angulartics2DSpace plugin is only active in the browser and not on the server.

### CODE REVIEW

**Comment by artlowel** [comment]
Travis is failing due to an issue with the default rest config used by the docker backend. We tried that docker backend locally, and it seems the dspace.restUrl param isn't set. You can also see that in the output of some of the e2e tests that _do_ succeed. E.g. [here](https://travis-ci.org/DSpace/dspace-angular/jobs/597514899#L5505) it says:
```

Http failure response for http://localhost:8080/api/dso/find?uuid=a7e444e5-594b-42e3-9a39-7cb4d9b5e325
```
That URL should be ht<span>tp://</span>localhost:8080/<b>server</b>/api/…`

This commit fixes that issue: https://github.com/antoine-atmire/DSpace/commit/3ac6ca51b940d41f9893f86f85099cffb754c2d7

We haven't created a PR because there may be reasons not to set this on rest repo, we may just want to set it for the docker image. In any case, afterwards the docker image will need to be rebuilt

**Comment by tdonohue** [comment]
If you rebase on latest `master`, I believe the e2e testing issues are now fixed (see #500 which was just merged).

File: src/app/+search-page/search.component.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import { getSucceededRemoteData } from '../core/shared/operators';
+import { RouteService } from '../core/services/route.service';
+import { SEARCH_CONFIG_SERVICE } from '../+my-dspace-page/my-dspace-page.component';
+
+export const SEARCH_ROUTE = '/search';
+
+/**
+ * This component renders a simple item page.
+ * The route parameter 'id' is used to request the item it represents.
+ * All fields of the item that should be displayed, are defined in its template.
```
Comment (tdonohue) [inline src/app/+search-page/search.component.ts]: This TypeDoc comment needs updating.  This is the Search component, not an Item page.

**Comment by antoine-atmire** [comment]
@tdonohue all set

**Comment by tdonohue** [comment]
@antoine-atmire : After merging your other PR (#513), this one has addition merge conflicts to be resolved.

**Comment by tdonohue** [comment]
As this is at +2, merging it!  Thanks again @antoine-atmire for your hard work here.

**Pull request merged**